.github/workflows/documentation.yml

```diff
@@ -44,62 +51,58 @@ jobs:
         uses: ocaml/setup-ocaml@v3
         with:
           ocaml-compiler: ${{ env.OCAML_DEFAULT_VERSION }}
-          dune-cache: true
+          # https://github.com/ocaml/dune/issues/7720
+          dune-cache: false
+
+      - name: Retrieve annotated tags for dune
+        run: git fetch --tags --force
+
 
       # Install dependencies if the cache is not retrieved
       # odoc is installed as deps with { with-doc } in opam files
       - name: opam install deps
         run: opam exec -- make deps
-      # if: steps.cache-opam.outputs.cache-hit != 'true'
 
-      # Try to upgrade installed packages and fix dependencies if needed,
-      # when the cache is retrieved
-      # - run: opam upgrade --fixup
-      #  if: steps.cache-opam.outputs.cache-hit == 'true'
+      - name: Perform version substitution
+        run: opam exec -- dune subst
 
       # Make the documentation
       - name: Make OCaml documentation
-        run: opam exec -- make odoc
+        run: opam exec -- dune build @doc
 
       # Create and upload an artifact `ocaml_doc` that contains the ocaml
       # documentation in html format.
-      # This is only done if this workflow is triggered in a PR or on the
-      # following branches : next, main
       - name: Upload ocaml documentation
         uses: actions/upload-artifact@v4
-        if: github.event_name == 'pull_request' || github.ref == 'refs/heads/next' || github.ref == 'refs/heads/main'
         with:
           name: ocaml_doc
           path: _build/default/_doc/_html/
 
-
-  # On PR, or push on next/main, build the sphinx general documentation
-  # If this build fails, the documentation workflow stops
-  # If it succeeds, an artifact is made with the generated documentation
-  # This artifact is used in the deploying job
   sphinx_docs:
     name: Sphinx documentation
-
-    # We only run this if the ocaml documentation build is successful
-    needs: ocaml_docs
-
-    # Sphinx documentation is only builded if a PR is open or when it's
-    # pushed on next or main
-    if: github.event_name == 'pull_request' || github.ref == 'refs/heads/next' || github.ref == 'refs/heads/main'
     runs-on: ubuntu-latest
 
+    outputs:
+      version-name: ${{ steps.version-name.outputs.target }}
+
     steps:
       # Checkout the code of the current branch
       - name: Checkout code
         uses: actions/checkout@v4
 
+      - name: Determine version name
+        id: version-name
+        run: |
+          echo "target=${GITHUB_REF#*/*/}" >> $GITHUB_OUTPUT
+
       # Build the sphinx documentation
       # and automatically print any error or warning
       - name: Build sphinx documentation
         uses: ammaraskar/sphinx-action@master
         with:
           docs-folder: "docs/sphinx_docs/"
-          build-command: "sphinx-build -b html . _build"
+          build-command: |
```

Comment:
We may add the flag `-W` to fail on warnings?

Reply:
Unfortunately not yet because we have warnings. We can do it once they are fixed.

Reply:
(The warning existed before this PR, it is related to licenses — I did not investigate)

Reply:
Ok let's merge and we will fix the warnings later.